packages/amplify-graphql-model-transformer/src/resources/amplify-dynamodb-table/amplify-table-manager-lambda/amplify-table-manager-handler.ts

```diff
@@ -74,40 +74,27 @@ export const onEvent = async (event: AWSCDKAsyncCustomResource.OnEventRequest):
       }
       log('Current table state: ', describeTableResult);
 
-      // determine if table needs replacement
+      // determine if table needs replacement when table key schema is changed
       if (isKeySchemaModified(describeTableResult.Table.KeySchema!, tableDef.keySchema)) {
-        console.log('Update requires replacement');
+        console.log('Detected table key schema change. Update requires replacement');
         if (tableDef.allowDestructiveGraphqlSchemaUpdates) {
-          if (describeTableResult.Table.DeletionProtectionEnabled === true) {
-            throw new Error('Table cannot be replaced when the deletion protection is enabled.');
-          }
-          console.log('Deleting the old table');
-          await ddbClient.deleteTable({ TableName: event.PhysicalResourceId });
-          await retry(
-            async () => await doesTableExist(event.PhysicalResourceId!),
-            (res) => res === false,
-          );
-          console.log(`Table '${event.PhysicalResourceId}' does not exist. Deletion is finished.`);
-
-          const createTableInput = toCreateTableInput(tableDef);
-          const response = await createNewTable(createTableInput);
-          result = {
-            PhysicalResourceId: response.tableName,
-            Data: {
-              TableArn: response.tableArn,
-              TableStreamArn: response.streamArn,
-              TableName: response.tableName,
-              IsTableReplaced: true, // This value will be consumed by isComplete handler
-            },
-          };
-          log('Returning result', result);
-          return result;
+          return replaceTable(describeTableResult.Table, tableDef);
         } else {
           throw new Error(
             "Editing key schema of table requires replacement which will cause ALL EXISITING DATA TO BE LOST. If this is intended, set the 'allowDestructiveGraphqlSchemaUpdates' to true and deploy again.",
           );
         }
       }
+      // when both sandbox and destructive updates are enabled, gsi update will trigger a table replacement
+      if (tableDef.replaceTableUponGsiUpdate && tableDef.allowDestructiveGraphqlSchemaUpdates) {
+        const nextUpdate = getNextGSIUpdate(describeTableResult.Table, tableDef);
+        if (nextUpdate !== undefined) {
```

Comment:
why do we replace the table only when we have a next update? I don't see where we're using this computed `nextUpdate`?

Reply:
The `nextUpdate` is a condition here, which indicates the GSI update detected. This is for a special use case for the initial schema design phase when you could do multiple GSI changes and do not care about data entries. The GSIs can be created much faster by using a create table command instead of update table command.